lib/web_ui/lib/src/engine/text_editing.dart

```diff
@@ -888,3 +900,45 @@ class _EditableSizeAndTransform {
   final double height;
   final Float64List transform;
 }
+
+/// Used for detecting the changes in the selection.
+///
+/// Currently only used in Firefox.
+///
+/// In Firefox, when cursor moves, nor selectionChange neither onInput
+/// events are triggered. We are listening to keyup event. Selection start,
+/// end values are used to decide if the cursor moved.
+///
+/// Specific keycodes are not checked since users/applicatins can bind their own
+/// keys to move the cursor.
+class SelectionChangeDetection {
+  int _start = 0;
+  int _end = 0;
```

Comment:
Once the constructor is modified to take a `domElement` as argument, the `start` and `end` values should be initialized correctly from the dom element.

Reply:
done.